mlrun/runtimes/nuclio/application/application.py

```diff
@@ -469,14 +488,22 @@ def create_api_gateway(
         api_gateway_scheme = db.store_api_gateway(
             api_gateway=api_gateway.to_scheme(), project=self.metadata.project
         )
-        if not self.status.api_gateway_name:
-            self.status.api_gateway_name = api_gateway_scheme.metadata.name
-        self.status.api_gateway = APIGateway.from_scheme(api_gateway_scheme)
-        self.status.api_gateway.wait_for_readiness()
-        self.url = self.status.api_gateway.invoke_url
 
-        logger.info("Successfully created API gateway", url=self.url)
-        return self.url
+        if set_as_default:
+            self.status.api_gateway_name = api_gateway_scheme.metadata.name
+            self.status.api_gateway = APIGateway.from_scheme(api_gateway_scheme)
+            self.status.api_gateway.wait_for_readiness()
+            self.url = self.status.api_gateway.invoke_url
+            url = self.url
+        else:
+            api_gateway = APIGateway.from_scheme(api_gateway_scheme)
+            api_gateway.wait_for_readiness()
+            url = api_gateway.invoke_url
+            # Update application status
+            self._get_state(raise_on_exception=False)
```

Comment:
why is this needed here?

Reply:
to get the external invocation. it is enriched only in the server but the client doesn't get it